pkg/api/converterfromapi.go

```diff
@@ -182,10 +190,18 @@ func convertPropertiesToV20160330(api *Properties, p *v20160330.Properties) {
 		convertMasterProfileToV20160330(api.MasterProfile, p.MasterProfile)
 	}
 	p.AgentPoolProfiles = []*v20160330.AgentPoolProfile{}
-	for _, apiProfile := range api.AgentPoolProfiles {
-		v20160330Profile := &v20160330.AgentPoolProfile{}
-		convertAgentPoolProfileToV20160330(apiProfile, v20160330Profile)
-		p.AgentPoolProfiles = append(p.AgentPoolProfiles, v20160330Profile)
+	// We expect exactly 2 agent pools: index [0] is the private pool, index [1] is the public pool
+	for i, apiProfile := range api.AgentPoolProfiles {
+		// a V20160330 should only have one (private) agent pool
+		if i < 1 {
+			v20160330Profile := &v20160330.AgentPoolProfile{}
```

Comment:
we would have to add it... I was thinking more of checking their names match what we set when splitting them/only one has a dns prefix/ validating there are only 2 agent pool profiles. It would be less secure then marking them somehow. mostly just thinking out loud at this point